Improve SignalR connectivity feedback

StoyanGoranov opened the pull request:
Related to: https://github.com/Not-Endurance/not-timing-system/issues/364 
but doesn't close it since issue also includes changes to EMS.

StoyanGoranov committed:
add relevant propeties to ConnectionBehind state

https://github.com/Not-Endurance/not-timing-system/issues/364

achobanov commented on src/Judge/NTS.Judge/Core/Behinds/Adapters/ConnectionsBehind.cs:
```diff
@@ -24,6 +24,8 @@ public ConnectionsBehind(IRpcSocket rpcSocket)
 
     protected override Task<bool> PerformInitialization(params IEnumerable<object> arguments)
     {
+        ServerConnectionStatus = RpcConnectionStatus.Disconnected;
+        IsServerConnected = false;
```
I don't think these lines are necessary. Disconnected and false are both default values for their types. Which is not consistent actually with how we do enums - they should not contain 0 as value. But keep RpcConnectionStatus as is for now, we'll revisit once we do the NTS Witness.

achobanov replied:
Furthermore is `IsServerConnected` simply checking the connection status then make it readonly:
```c#
public bool IsServerConnected => ServerConnectionStatus == RpcConnectionStatus.Connected;
```

StoyanGoranov replied:
Initializing the `ServerConnectionStatus` and `IsServerConnected` properties in `PerformInitialization` ensures they are part of the Behind state. This is needed because the properties are used in the `ConnectionStatus` component and changes to the status of the connection are reflected by the UI this way. 

This effectively eliminates the need to introduce a Timer which checks the status of the connection every second.

StoyanGoranov replied:
Setting the IsServerConnected value is needed to ensure it is part of the `ConnectionsBehind` state.

achobanov replied:
Remove it and you'll see it will still be working.

StoyanGoranov committed:
simplify IsServerConnected property at ConnectionsBehind

https://github.com/Not-Endurance/not-timing-system/issues/364

StoyanGoranov committed:
inherit IStartupInitializer and ISingleton at IRpcClient

needed to ensure the RpcClient is initialized when remote procedure
calls are received

https://github.com/Not-Endurance/not-timing-system/issues/364

StoyanGoranov committed:
implement IStartupInitializer at Connections- and ParticipationClient

https://github.com/Not-Endurance/not-timing-system/issues/364

StoyanGoranov committed:
move RpcSocket.Connect call after initializers run

ensuring procedures are registered before RpcSocketConnect is called, not after

https://github.com/Not-Endurance/not-timing-system/issues/364

StoyanGoranov committed:
Revert "simplify IsServerConnected property at ConnectionsBehind"

This reverts commit 4073285130f6cab04a85655bca367d8491fb5cc8.

StoyanGoranov committed:
clean-up ConnectionsBehind PerformInitialization

StoyanGoranov committed:
hide Reconnect button when RfidDetection is disabled

https://github.com/Not-Endurance/not-timing-system/issues/364

StoyanGoranov committed:
Merge branch 'develop' into improve-signalr-connectivity-feedback

github-actions[bot] committed:
Apply Csharpier code formatting

achobanov committed:
Merge branch 'develop' into improve-signalr-connectivity-feedback

achobanov committed:
Merge branch 'develop' into improve-signalr-connectivity-feedback
